django_select2/forms.py

```diff
@@ -175,17 +175,28 @@ def value_from_datadict(self, data, files, name):
 class HeavySelect2Mixin(object):
     """Mixin that adds select2's AJAX options and registers itself on Django's cache."""
 
+    dependent_fields = {}
+
     def __init__(self, **kwargs):
         """
         Return HeavySelect2Mixin.
 
         Args:
             data_view (str): URL pattern name
             data_url (str): URL
-
+            dependent_fields (List[str]): List of dependent parent fields.
+                The value of the dependent field will be passed as to :func:`.filter_queryset`.
+                It can be used to further restrict the search results. For example, a city
+                widget could be dependent on a country.
         """
         self.data_view = kwargs.pop('data_view', None)
         self.data_url = kwargs.pop('data_url', None)
+
+        dependent_fields = kwargs.pop('dependent_fields', None)
+        if dependent_fields is not None:
+            self.dependent_fields = dependent_fields
```

Comment:
Just cast as a dict here. If the type is not dict or a list of tuples, it will fails, which is enough developer feedback.